agilent_33120a_function_gen.py

```diff
@@ -38,25 +38,72 @@
 from struct import unpack
 from twisted.internet.defer import inlineCallbacks, returnValue
 from labrad.units import Unit, Value, V
+import labrad.units as U
 
 class AgilentFunctionGenerator(GPIBManagedServer):
     name = 'Agilent 33120a generator'
     deviceName = 'HEWLETT-PACKARD 33120A'
-    @setting(11,'Set DC', val='v[V]', returns='')
-    def set_dc_waveform(self, c,val=0*V):
-        if val< -5 or val > 5:
-            raise Exception('Signal Gnerator only puts out -5 to 5 volts in DC Voltage')
-        """Puts generator into DC mode with given voltage."""
+    
+    @setting(11, 'clear')      
+    def clear(self, c):
+        """Clears status byte summary and event registers
+
+        """
+
         dev = self.selectedDevice(c)
-        dev.write(':APPL:DC DEF, DEF, %f' % float(val))
+        dev.write('*CLS')
       
-    @setting(12, 'set impedance', setting='s', returns='')      
-    def set_impedance(self, c, setting = '50'):
+    @setting(12, 'set impedance', val='s', returns='')      
+    def set_impedance(self, c, val = '50'):
+        """Sets the loading impedance
+        
+        args: 
+           val (str): The loading impedance
+        """
+
         allowed = ['50', 'INF']
         if setting not in allowed:
             raise Exception('allowed settings are: %s' % allowed)
         dev = self.selectedDevice(c)
-        dev.write('OUTP:LOAD %s' % setting)
+        dev.write('OUTP:LOAD {}'.format(val))
+
+    @setting(13,'Set DC voltage', val='v[V]', returns='')
```

Comment:
You changed the name here (was "Set DC") and the setting id. The id is generally not a problem because we call these by name, but changing the name could break people's code. Is there a reason for the change?